### ISSUE TITLE
Comments of the form /*__CSSVARSPONYFILL-XY__*/ are written into style tags

### ISSUE THREAD

**Issue by pseudoramble** [jhildenbiddle/css-vars-ponyfill#15]
When working on Issue #13 it was noted that strange comment was left behind within the style tag. They look like this:

```
/*__CSSVARSPONYFILL-01__*//*__CSSVARSPONYFILL-03__*//*__CSSVARSPONYFILL-05__*/...
:root {
  --font: 25px;
}
...
/*__CSSVARSPONYFILL-07__*//*__CSSVARSPONYFILL-09__*//*__CSSVARSPONYFILL-11__*/
```

In other words, the strange comments are left within the style tag and wrap around the actual styles to be controlled via the ponyfill.

**Comment by pseudoramble** [comment]
From the investigation I was able to do so far, it looks like all of the evenly-numbered values are being replaced as expected, and the odd values are being left behind. This looks like it's because the `cssText` variable is being used to match on results and also as the final output. Since `cssMatches` is maintained throughout the iteration to perform the match, it keeps track of what index to start from, and then accidentally skips over variables on each step.

A brief example - Say there are 7 entries in `cssArray` where the entry at index 3 has some actual CSS. `cssText` then has the value `/*-_CSSVARSPONYFILL-00*//*-_CSSVARSPONYFILL-01*//*-_CSSVARSPONYFILL-02*/:root{ --font: 25px } .some-class{ font-size: var(--fontSize); font-weight: 800; font-family: Arial; }/*-_CSSVARSPONYFILL-04*//*-_CSSVARSPONYFILL-05*//*-_CSSVARSPONYFILL-06*/` to start with.

Iteration 1:

```
// cssMatch.lastIndex = 0
// cssText.length = 246
cssText = cssText.replace(cssMatch[0], cssArray[i]);
// -00 has been removed
```

Iteration 2:

```
// cssMatch.lastIndex = 25
// cssText.length = 222
cssText = cssText.replace(cssMatch[0], cssArray[i]);
// -02 has been removed because it starts matching at char 25
```
Iteration 3:

```
// cssMatch.lastIndex = 50
// cssText.length = 174
cssText = cssText.replace(cssMatch[0], cssArray[i]);
// -04 has been removed because it starts matching at char 50 which lands in the middle of the CSS
```

Iteration 4:

```
// cssMatch.lastIndex = 75
// cssText.length = 150
cssText = cssText.replace(cssMatch[0], cssArray[i]);
// -06 has been removed because of where it starts matching
```

This is where the loop terminates because there is no more match beyond this point.

**Comment by jhildenbiddle** [comment]
Very, very helpful, @pseudoramble. Thank you for taking the time to investigate.

Looks like this should be a relatively easy fix. I won’t be available to tackle this until EOW, so an update should be ready early next week.

Thanks again!

**Comment by jhildenbiddle** [comment]
Fixed in 1.9.0, with automated tests added to the build.

Thanks again for filing the issue! Really appreciate the help tracking this one down, @pseudoramble. 😄
